src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,176 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> previous;
+
+        private Node(Node<T> previous, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.previous = previous;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
```

Comment:
```suggestion
		Node<T> newNode = new Node<>(tail, value, null);
        if (size == 0) {
            head = newNode;
        } else {
            tail.next = newNode;
        }
        tail = newNode;
```

It seems to me it looks simpler that way

Reply:
I agree.